src/vm/exceptionhandling.cpp

```diff
@@ -441,13 +441,24 @@ void ExceptionTracker::UpdateNonvolatileRegisters(CONTEXT *pContextRecord, REGDI
         }                                                                                   \
     } while (0)
 
+#define UPDATEREG_VAL(reg)                                                                  \
+    do {                                                                                    \
+        STRESS_LOG2(LF_GCROOTS, LL_INFO100, "Updating " #reg " %p to %p\n",                 \
+                pContextRecord->reg,                                                        \
+                pRegDisplay->pCurrentContext->reg);                                         \
+        pContextRecord->reg = pRegDisplay->pCurrentContext->reg;                            \
+    } while (0)
+
+
 #if defined(_TARGET_X86_)
 
     UPDATEREG(Ebx);
     UPDATEREG(Esi);
     UPDATEREG(Edi);
     UPDATEREG(Ebp);
 
+    UPDATEREG_VAL(ResumeEsp);
```

Comment:
Great, that sounds much better to me! And maybe we can just change the method to set the ResumeEsp internally in the context instead of returning it back. And maybe rename the method to something like FixResumeEsp then.